lib/rack/idempotency.rb

```diff
@@ -19,23 +20,47 @@ class Idempotency
     def initialize(app, store: NullStore.new)
       @app     = app
       @store   = store
+      if @store.class.to_s == "Rack::Idempotency::RedisStore"
+        @mutex_mode = true
+        @cache_errors = true
```

Comment:
Whether we want to cache errors or not is totally independent from which store we use (or whether the store supports mutex lock). This does not belong here.

Reply:
I do agree, cache errors was here only because, it changes default/wrong behavior of the gem. Updating spec in case or error response should be better.